plugins/modules/nmcli.py

```diff
@@ -2384,6 +2384,25 @@ def down_connection(self):
         cmd = [self.nmcli_bin, "con", "down", self.conn_name]
         return self.execute_command(cmd)
 
+    def get_connection_state(self):
+        """Get the current state of the connection"""
+        cmd = [self.nmcli_bin, "--terse", "--fields", "GENERAL.STATE", "con", "show", self.conn_name]
+        (rc, out, err) = self.execute_command(cmd)
+        if rc != 0:
+            raise NmcliModuleError(err)
+
+        lines = [ll.strip() for ll in out.splitlines() if "GENERAL.STATE" in ll]
+        for line in lines:
+            if "GENERAL.STATE" in line:
+                state = line.split(":")
```

Comment:
In your first submission, you had coded this as:
```
                state = line.split(":")[-1].strip()
```
Being `[-1]` (the last item of the split), the `strip()` becomes redundant as we have performed `strip()` in all lines (2394).

However, while previously you were splitting the line and returning ONLY the last element (`[-1]`), now you are returning the complete list of results from the split.

One of these two is bound to be wrong. Could you please confirm which one is the correct one?